astropy/table/table.py

```diff
@@ -732,6 +732,10 @@ def __init__(
         # a list of dict. If data are not list of dict then this is None.
         names_from_list_of_dict = None
 
+        # Treat empty rows same as None case.
+        if isinstance(rows, (list, tuple)) and len(rows) < 1:
```

Comment:
Style nit
```suggestion
        if isinstance(rows, (list, tuple)) and not rows:
```
if you really want very explicit checks, I'd still prefer this:
```suggestion
        if isinstance(rows, (list, tuple)) and len(rows) == 0:
```

More importantly, the check doesn't seem general enough as lets empty numpy arrays through.
Generalizing the `isinstance` check might not be the right approach: I think the would-be correct check is `isinstance(rows, typing.Sequence)`, but unfortunately [this isn't supported by numpy](https://github.com/numpy/numpy/issues/2776). So how about we keep it extremely simple, e.g. 
```suggestion
        if not rows:
```
?

Reply:
I'll defer to @taldcroft on which is preferred here.